### PR TITLE
Normalize 00 prefix, handle Lucerne numbers

### PR DESCRIPTION
**Pull request by cb341**
[TICKET-23811](https://redmine.renuo.ch/issues/23811)

According to the logs, sometimes we receive calls like `0041*********`. These should be properly normalized like the rest.
I also decided to add more edge cases to test against for more certainty.

We don't have 3CX logs that show that the double prefixed variant of a phone number lead to a contact number not being found. We have some instances where even a properly normalized number would not be found. Either way the normalizer handles this case **incorrectly** at the moment.

Although not explicitly stated in the ticket,  I noticed that we errornously remove the 041 prefix for Luzerne numbers.
This could lead to problems so I updated the normalization. Didn't find an instance of it in our logs but it is another case I'd like to avoid.

### CODE REVIEW

File: app/models/contact_serializer.rb [hunk 1]
```diff
@@ -22,7 +22,17 @@ def map_phone_numbers_to_keys(phone_numbers)
     end
 
     def normalize_phone_number(phone_number)
-      phone_number.gsub(/[^+0-9]/, "").gsub(/^[+0]41/, "").rjust(10, "0")
+      n = phone_number.to_s.gsub(/[^0-9]/, "")
+      # Strip 0041, 041, or 41 prefixes IF the number is long enough to include a country code.
+      # "n.length > 10" protects Lucerne numbers (e.g. 041 222 33 44), which are exactly 10.
```
Comment (cb341) [inline app/models/contact_serializer.rb]: Didn't know in which better way I could express myself here other than in a comment.

**Review by sislr — APPROVED** [review]
